### ISSUE TITLE
Error adding variable support to a data source plugin

### ISSUE THREAD

**Issue by mure** [grafana/grafana#44738]
**What happened**:
I am trying to migrate my data source plugin to use the new API from here: https://github.com/grafana/grafana/pull/28217
I want to extend from 'DataSourceVariableSupport', because my understanding is that it will reuse the QueryEditor I've already defined when creating query variables.

I added this code:
_variables.ts_
```ts
export class MyVariableSupport extends DataSourceVariableSupport<MyDataSource, MyQuery> {
}
```
_DataSource.ts_
```ts
constructor(instanceSettings: DataSourceInstanceSettings<MyDataSourceOptions>) {
  ...
  this.variables = new MyVariableSupport();
}
```

After building my plugin and trying to add a new query variable in the UI, I get this error toast:

MyPlugin plugin failed
TypeError: Class constructor we cannot be invoked without 'new'

In the console, I see this error: `{message: 'Datasource: 0LO7l-x7k was not found'}`

**What you expected to happen**: QueryEditor should appear in the Variables>Edit view and there shouldn't be an error.

**Environment**:
- Grafana version: 8.4.0
- OS Grafana is installed on: MacOS
- User OS & Browser: MacOS + Chrome

**Comment by mure** [comment]
Hmm...I think I'm actually hitting this issue: https://github.com/grafana/grafana/pull/32664 Is there something different about how the new APIs are bundled?

**Comment by zuchka** [comment]
Thanks for creating this issue, @mure 

these are great questions for the Plugin Development category on our forum:

https://community.grafana.com/c/plugin-development/30

**Comment by mure** [comment]
@zuchka
There's behavior here that needs to be addressed. If you create a datasource plugin using grafana-toolkit and then try to add query variable support, your plugin will fail to load with an unhelpful error.

**Comment by zuchka** [comment]
no worries! reopened.

the plugin folks might want to see what devDependencies and dependencies are listed in your `package.json` -- especially what version of `toolkit` you are using 👍

**Comment by mckn** [comment]
This will be resolved by the following fix available in Grafana 8.5 and forward https://github.com/grafana/grafana/pull/44842
